tests/cupy_tests/random_tests/test_permutations.py

```diff
@@ -1,9 +1,67 @@
 import unittest
 
+import numpy
+
+import cupy
 from cupy import testing
 
 
 @testing.gpu
 class TestPermutations(unittest.TestCase):
 
     _multiprocess_can_split_ = True
+
+
+@testing.gpu
+class TestShuffle(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    # Test ranks
+
+    @testing.numpy_cupy_raises()
+    def test_shuffle_zero_dim(self, xp):
+        a = testing.shaped_random((), xp)
+        xp.random.shuffle(a)
+
+    def test_shuffle_two_or_more_dim(self):
+        a = testing.shaped_random((2, 3), cupy)
+        with self.assertRaises(TypeError):
+            return cupy.random.shuffle(a)
+
+    # Test dtypes
+
+    @testing.for_dtypes([numpy.float16, numpy.bool_])
```

Comment:
I've implemented `numpy.bool_` case. But I keep `numpy.float16` case as future work because it takes long time.